save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationContestsMenu.kt

```diff
@@ -0,0 +1,164 @@
+
+import react.dom.html.ButtonType
+import react.dom.html.ReactHTML
+import react.dom.html.ReactHTML.button
+import react.dom.html.ReactHTML.div
+import react.table.columns
+
+/**
+ * CONTESTS tab in OrganizationView
+ */
+val organizationContestsMenu = organizationContestsMenu()
+
+@Suppress("MAGIC_NUMBER", "TYPE_ALIAS")
+private val contestsTable: FC<OrganizationContestsTableProps<ContestDto>> = tableComponent(
+    columns = columns {
+        column(id = "name", header = "Contest Name", { name }) { cellProps ->
+            Fragment.create {
+                ReactHTML.td {
+                    ReactHTML.a {
+                        href = "#/contests/${cellProps.row.original.name}"
+                        +cellProps.value
+                    }
+                }
+            }
+        }
+        column(id = "description", header = "Description", { description }) { cellProps ->
+            Fragment.create {
+                ReactHTML.td {
+                    +(cellProps.value ?: "Description is not provided")
+                }
+            }
+        }
+        column(id = "start_time", header = "Start Time", { startTime.toString() }) { cellProps ->
+            Fragment.create {
+                ReactHTML.td {
+                    +cellProps.value.replace("T", " ")
+                }
+            }
+        }
+        column(id = "end_time", header = "End Time", { endTime.toString() }) { cellProps ->
+            Fragment.create {
+                ReactHTML.td {
+                    +cellProps.value.replace("T", " ")
+                }
+            }
+        }
+    },
+    initialPageSize = 10,
+    useServerPaging = false,
+    usePageSelection = false,
+    getAdditionalDependencies = {
+        arrayOf(it.isContestCreated)
+    }
+)
+
+/**
+ * OrganizationContestsMenu component props
+ */
+external interface OrganizationContestsMenuProps : Props {
+    /**
+     * Current organization name
+     */
+    var organizationName: String
+
+    /**
+     * [Role] of user that is observing this component
+     */
+    var selfRole: Role
+
+    /**
+     * Callback to show error message
+     */
+    var updateErrorMessage: (Response) -> Unit
+}
+
+/**
+ * Interface for table reloading after the contest creation
+ */
+external interface OrganizationContestsTableProps<D : Any> : TableProps<D> {
+    /**
+     * Flag to update table data when contest is created
+     */
+    var isContestCreated: Boolean
+}
+
+@Suppress(
+    "TOO_LONG_FUNCTION",
+    "LongMethod",
+    "MAGIC_NUMBER",
+    "ComplexMethod"
+)
+private fun organizationContestsMenu() = FC<OrganizationContestsMenuProps> { props ->
+    val (isToUpdateTable, setIsToUpdateTable) = useState(false)
+    val (isContestCreationModalOpen, setIsContestCreationModalOpen) = useState(false)
+    showContestCreationModal(
+        props.organizationName,
+        isContestCreationModalOpen,
+        {
+            setIsContestCreationModalOpen(false)
+            setIsToUpdateTable { !it }
```

Comment:
What does this achieve? Table will be re-rendered once this flag is flipped, because it's a dependency.

Reply:
We need it in order to fetch data from backend only when a new contest is created. If we add `isContestCreationModalOpen` to be a dependency, we will fetch data each time the modal is opened/closed which is not good. Now we change this flag only on successful contest creation

Reply:
Please add a comment in code or do something like
```kotlin
val refreshTable = { setIsToUpdateTable { !it } }
```

Reply:
Done